lib/SerialIO.js

```diff
@@ -16,61 +16,65 @@ class SerialIO {
     this.d('initializing SerialIO on port: %s', port)
     this._sp = sp
     this._portString = port
-    this._port = new this._sp(port, { autoOpen: false })
+
+    /**
+     * Indicates if a message transaction is in progress
+     * @type {boolean}
+     */
+    this.sending = false
+
+    /**
+     * indicates if closing event is intended behaviour
+     * @type {boolean}
+     * @private
+     */
+    this._closing = false
+
+    /**
+     * Holds event handlers
+     * @type {Object.<string, function>}
+     * @private
+     */
+    this._handlers = {}
+  }
+
+  /**
+   * Prepares the serialport (without opening it). May throw an error, if the target port does not exist
+   */
+  preparePort () {
+    this._port = new this._sp(this._portString, { autoOpen: false })
     this._parser = new DataParser()
     this._parser.onMessage((msg) => this._handleMessage(msg))
```

Comment:
A cleaner way to write the same thing:
`this._parser.onMessage(this._handleMessage)`

Reply:
as this discussed, this would change the execution context of `this._handleMessage`, which breaks the code.